backend/tournesol/tests/test_text_search.py

```diff
@@ -0,0 +1,712 @@
+from django.test import TestCase
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from tournesol.entities.candidate import TYPE_CANDIDATE
+from tournesol.entities.video import TYPE_VIDEO
+from tournesol.models import Poll
+from tournesol.tests.factories.entity import (
+    EntityFactory,
+    UserFactory,
+    VideoCriteriaScoreFactory,
+    VideoFactory,
+)
+from tournesol.tests.factories.entity_score import EntityCriteriaScoreFactory
+from tournesol.tests.factories.poll import PollWithCriteriasFactory
+from tournesol.tests.factories.ratings import (
+    ContributorRatingCriteriaScoreFactory,
+    ContributorRatingFactory,
+)
+
+
+class TextSearchTestCase(TestCase):
+    """
+    The text search can be used by multiple recommendations views,
+    such as polls recommendations, and contributor recommendations.
+    It can also be used for multiple polls.
+
+    Its purpose is to filter some results based on textual input.
+    It is currently based on the PostgreSQL full-text search.
+
+    The tests are based on many design choices, that are likely to
+    change over time.
+    """
+
+    def setUp(self):
+        self.poll = Poll.default_poll()
+
+        self.query = "fruit"
+
+        self.field = "10 best fruit cake recipes"

```

Comment:
Since there are numerous occurrences, I'll use self.text. 
I'll maybe keep the term "field" for _create_rated_entity, making it more explicit that it is metadata fields. Or do you think it's better to also use "text1", "text2", .... ?

Reply:
I think it's good to use `self.text` as an attribute of the test case, and I agree with you to keep the abstract `field` arguments in the `_create_rated_entity` method as it is used to fill different fields depending on the entity created.